enclave/crypto/cert.c

```diff
@@ -764,8 +764,8 @@ oe_result_t oe_cert_chain_read_pem(
     result = OE_OK;
 
 done:
-
-    _referent_free(referent);
+    if (referent)
+        _referent_free(referent);
```

Comment:
Same comment here about _referent_free(), it should a no-op if NULL is passed so this change should not be needed.  Fix the header so no warning is generated.

Reply:
If you enter from mbedtls_x509_crt_free(referent->crt) and referent is a null pointer, then access to field crt in _referent_free() results in a dereference of a null pointer, therefore the warning.

Reply:
_referent_free does not check for a null pointer. We should fix the implementation to check there.

Reply:
Okay. Working on it now.

Reply:
+1 to fixing _referent_free to check for a null pointer, for consistent with the rest of the *free() functions